tests/Fixtures/Test/FileReaderTest/StdinFakeStream.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixtures\Test\FileReaderTest;
+
+/**
+ * @author ntzm
+ *
+ * @internal
+ *
+ * @see \PhpCsFixer\Tests\FileReaderTest::testReadStdin
+ *
+ * A stream wrapper class that pretends to be php://stdin, and returns:
+ * - <?php echo "foo";
+ * - <?php echo "bar";
```

Comment:
this is not how `php://stdin` will behave for us:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.11/src/StdinFileInfo.php#L34

Reply:
How else to test that it caches?
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.2/src/FileReader.php#L56-L600

Reply:
I knew that about POST requests in HTTP. So this is true about CLI as well?

Reply:
Ah OK I get you now, will update, thanks

Reply:
Woops, I misread the doc: it tells `php://input` is not (always?) seekable prior to PHP 5.6 but I couldn't find documentation regarding `php://stdin` not being seekable.

Reply:
```
ker@dus:~/github/FriendsOfPHP/PHP-CS-Fixer λ cat x.php 
<?php

var_dump([
    file_get_contents('php://stdin'),
    file_get_contents('php://stdin'),
]);
ker@dus:~/github/FriendsOfPHP/PHP-CS-Fixer λ echo "foo" | php -f x.php
array(2) {
  [0]=>
  string(4) "foo
"
  [1]=>
  string(0) ""
}
```

Reply:
Seems to be seekable in all PHP 5+ versions https://3v4l.org/PoJlP

Reply:
well, even iin your link,
2nd run on stdin is seekable under 5.6 but not 5.5.
also, `file_get_contents` is not rewind, eg on 7.0

Reply:
That means this custom stream wrapper class could be dropped on 2.11 branch, right?

Reply:
Nope, `php://stdin` does not seem to be seekable at all on PHP 7.2:

```console
 $ cat test.php
<?php

$stdin1 = fread(STDIN, 3);
fseek(STDIN, 0);
$stdin2 = fread(STDIN, 3);

var_dump($stdin1, $stdin2);

 $ echo "foo" | php test.php
PHP Warning:  fseek(): stream does not support seeking in test.php on line 4
string(3) "foo"
string(1) "
"
```

Reply:
Weird.. So the behaviour wanted is after returning the content just to return an empty string forever?

Reply:
the behaviour wanted in php? no, but it works like that
the behaviour wanted in mock? yes, to mimic php one